src/utils/hideAndRepeats.ts

```diff
@@ -221,6 +226,8 @@ function getVisibleElements(
   elementTypes: string[] = [],
   repeat = false
 ) {
+  const [repeatKey] = getRepeatParentInfo(step);
```

Comment:
this function is running a lot of unnecessary functionality here to get numRepeats, which isn't used here. I think this would be better as it was before, and you can get repeatKey manually here.